third_party/xla_client/pjrt_computation_client.cc

```diff
@@ -183,9 +183,10 @@ std::optional<xla::OpSharding> PjRtComputationClient::GetDataSharding(
 
 std::vector<ComputationClient::DataPtr> PjRtComputationClient::TransferToServer(
     absl::Span<const TensorSource> tensors) {
-  metrics::TimedSection timed(TransferToServerMetric());
-  tsl::profiler::TraceMe activity("PjRtComputationClient::TransferToServer",
-                                  tsl::profiler::TraceMeLevel::kInfo);
+  auto timed = std::make_shared<metrics::TimedSection>(TransferToServer());
```

Comment:
this is not enough, you need to set the `returned_future` and use the `onReady`, check https://github.com/pytorch/xla/commit/afe7bb6b6562edcd368055899e17c691824575ac. You also need a test.

Reply:
I'm not an expert but  thought Future is when we're waiting for the result from another process. All we're doing here is explicitly transferring objects one by one. So I thought we don't need those additional future related code.

Reply:
ignore my previous comment. I added the necessary future logic.